crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,103 @@
+use tui::symbols;
+use tui::text::Span;
+use tui::widgets::{Axis, Block, Borders, Chart, Dataset};
+
+/// A new WorkerTimeState is created everytime there is a #worker change event
+struct WorkerCountRecord {
+    pub at_time: SystemTime,
+    pub worker_count: u32,
+}
+
+pub struct ClusterOverviewChart {
+    /// The state of the worker at different times.
+    worker_records: Vec<WorkerCountRecord>,
+    /// The time for which the data is plotted for
+    view_size: Duration,
+}
+
+impl ClusterOverviewChart {
+    pub fn update(&mut self, overview: &CollectedOverview) {
+        self.worker_records.push(WorkerCountRecord {
+            at_time: SystemTime::now(),
+            worker_count: overview.worker_overviews.len() as u32,
+        });
+    }
+
+    pub fn draw(&mut self, rect: Rect, frame: &mut DashboardFrame) {
+        let workers_in_view = self.get_workers_in_view();
+        let max_workers_in_view = workers_in_view
+            .iter()
+            .max_by(|a, b| (a.1.cmp(&b.1)))
+            .map(|(_, count)| (*count as f64))
+            .unwrap_or(0.00);
+
+        let worker_counts: Vec<(f64, f64)> = workers_in_view
+            .iter()
+            .map(|x| (x.0 as f64, x.1 as f64))
+            .collect();
+        let datasets = vec![Dataset::default()
+            .name("workers_connected")
+            .marker(symbols::Marker::Braille)
+            .style(Style::default().fg(Color::White))
+            .data(&worker_counts)];
+
+        let chart = Chart::new(datasets)
+            .block(
+                Block::default()
+                    .title(Span::styled(
+                        "Worker Connection History",
+                        Style::default()
+                            .fg(Color::White)
+                            .add_modifier(Modifier::BOLD),
+                    ))
+                    .borders(Borders::ALL),
+            )
+            .x_axis(
+                Axis::default()
+                    .title("time ->")
+                    .style(Style::default().fg(Color::Gray))
+                    .bounds([
+                        (get_time_as_secs(SystemTime::now()) - self.view_size.as_secs()) as f64,
+                        get_time_as_secs(SystemTime::now()) as f64,
+                    ]),
+            )
+            .y_axis(
+                Axis::default()
+                    .style(Style::default().fg(Color::Gray))
+                    .bounds([0.0, max_workers_in_view])
+                    .labels(vec![
+                        Span::from(0.to_string()),
+                        Span::from(max_workers_in_view.to_string()),
+                    ]),
+            );
+        frame.render_widget(chart, rect);
+    }
+
+    /// Returns the worker connection chart data for the view size
+    fn get_workers_in_view(&self) -> Vec<(u64, u64)> {
+        self.worker_records
+            .iter()
+            .filter(|count| count.at_time > SystemTime::now() - self.view_size)
```

Comment:
Extract `SystemTime::now()` to a local variable so that it is the same for all workers.